src/playercards/customizable/SummonedServitorUpgradeSheet.ttslua

```diff
@@ -31,27 +31,41 @@ existingBoxes = { 1, 1, 1, 1, 1, 2, 3, 5 }
 
 inputBoxes = {}
 
+-- Locations of the slot selectors
+local SLOT_ICON_POSITIONS = {
+    arcane = { x = 0.160, z = 0.65},
+    ally = { x = -0.073, z = 0.65}
+  }
+  
+local selectedSlot = ""
+
 -- override 'marked boxes' for debugging ('all' or 'none')
 markDEBUG = ""
 
 -- save state when going into bags / decks
 function onDestroy() self.script_state = onSave() end
 
-function onSave() return JSON.encode({ markedBoxes, inputValues }) end
+function onSave() return JSON.encode({ markedBoxes, inputValues, selectedSlot }) end
 
 -- Startup procedure
 function onLoad(saved_data)
     if saved_data ~= "" and markDEBUG == "" then
         local loaded_data = JSON.decode(saved_data)
         markedBoxes = loaded_data[1]
         inputValues = loaded_data[2]
+        if #loaded_data > 2 then
+            selectedSlot = loaded_data[3]
+        else
+            selectedSlot = ""
+        end
```

Comment:
Since you now updated the save state, you could get rid of LOC 56 and LOC 58-60

Reply:
Fixed this incidentally while working on the importer code!  There is some code that might be worth cleaning up in some of the other sheets like Living Ink that have dead variables like `inputValues`, but it's more a clarity thing than functional